rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -328,10 +423,9 @@ wait(
       if (!(j < active_conditions->length())) {
         services->services[i] = 0;
       }
-      DDS::ReturnCode_t retcode = dds_wait_set->detach_condition(read_condition);
-      if (retcode != DDS::RETCODE_OK) {
-        RMW_SET_ERROR_MSG("Failed to get detach condition from wait set");
-        return RMW_RET_ERROR;
+      rmw_ret_t ret_code = __detach_condition(dds_wait_set, read_condition);
+      if (ret_code != RMW_RET_OK) {
```

Comment:
Why did we remove all the error messages from here and below?

Reply:
Not sure (wasn't the implementor). @ross-desmond  @mm318 ?

Reply:
Moved into another function (in person review)